src/_pytest/main.py

```diff
@@ -205,7 +205,7 @@ def wrap_session(config, doit):
             raise
         except Failed:
             session.exitstatus = EXIT_TESTSFAILED
-        except KeyboardInterrupt:
+        except (KeyboardInterrupt, exit.Exception):
```

Comment:
this implies that this is a breaking change for externals

Reply:
What do you suggest?
Should we have a custom exception that is still derived from KeyboardInterrupt (if possible)?

Reply:
@blueyed its not clear to me off hand, i wonder if having multiple base classes is acceptable while preparing for a breaking release (i dont see a way to deal out warnings about this one tho)

Reply:
Ok.
I've just noticed that also deriving from KeyboardInterrupt goes against what this tries to do: pdb not catching pytest's Exit.

I think it is OK for a feature release, and plugins should really catch/handle `Exit` in the first place already.

Reply:
> I think it is OK for a feature release, and plugins should really catch/handle Exit in the first place already.

I think so too. I don't think this will bring problems to anyone TBH.